sdks/java/io/jdbc/src/test/java/org/apache/beam/sdk/io/jdbc/JdbcIOAutoPartitioningIT.java

```diff
@@ -0,0 +1,403 @@
+import org.apache.beam.sdk.transforms.Count;
+import org.apache.beam.sdk.transforms.MapElements;
+import org.apache.beam.sdk.transforms.SimpleFunction;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.Row;
+import org.apache.beam.sdk.values.TypeDescriptor;
+import org.apache.beam.sdk.values.TypeDescriptors;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.collect.Lists;
+import org.joda.time.DateTime;
+import org.joda.time.Duration;
+import org.joda.time.Instant;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.model.Statement;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.JdbcDatabaseContainer;
+import org.testcontainers.containers.MySQLContainer;
+import org.testcontainers.containers.PostgreSQLContainer;
+
+/** A test of {@link org.apache.beam.sdk.io.jdbc.JdbcIO} on test containers. */
+@RunWith(Parameterized.class)
+public class JdbcIOAutoPartitioningIT {
+  private static final Logger LOG = LoggerFactory.getLogger(JdbcIOAutoPartitioningIT.class);
+
+  public static final Integer NUM_ROWS = 1_000;
+  public static final String TABLE_NAME = "baseTable";
+
+  @ClassRule public static TestPipeline pipelineWrite = TestPipeline.create();
+  @Rule public TestPipeline pipelineRead = TestPipeline.create();
+
+  @Parameterized.Parameters(name = "{0}")
+  public static Iterable<String> params() {
+    return Lists.newArrayList("mysql", "postgres");
+  }
+
+  @Parameterized.Parameter(0)
+  public String dbms;
+
+  public static JdbcDatabaseContainer<?> getDb(String dbName) {
+    if (dbName.equals("mysql")) {
+      return mysql;
+    } else {
+      return postgres;
+    }
+  }
+
+  // We need to implement this retrying rule because we're running ~18 pipelines that connect to
+  // two databases in a few seconds. This may cause the databases to be overwhelmed, and reject
+  // connections or error out. By using this rule, we ensure that each pipeline is trued twice so
+  // that flakiness from databases being overwhelmed can be managed.
+  @Rule
+  public TestRule retryRule =
+      new TestRule() {
+        // We establish max number of retries at 2.
+        public final int maxRetries = 2;
+
+        @Override
+        public Statement apply(Statement base, Description description) {
+          return new Statement() {
+            @Override
+            public void evaluate() throws Throwable {
+              Throwable caughtThrowable = null;
+              // implement retry logic here
+              for (int i = 0; i < maxRetries; i++) {
+                try {
+                  pipelineRead.apply(base, description);
+                  // base.evaluate();
```

Comment:
Be careful @pabloem , with this commented no tests are actually running. Everything just succeeds...
Almost missed this, but was caught by surprise to see the tests for string partitioning pass.

Reply:
oh wow incredible catch! thanks so much @mosche - I am so embarrassed that I had left this like that. Thanks!